fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/QueryTransformer.java

```diff
@@ -98,16 +98,8 @@ public LogicalPlan plan(SelectRelation queryBlock, ExpressionMapping outer) {
         builder = window(builder, analyticExprList);
 
         if (queryBlock.hasOrderByClause()) {
-            if (!queryBlock.hasAggregation()) {
-                //requires both output and source fields to be visible if there are no aggregations
-                builder = projectForOrder(builder,
-                        Iterables.concat(queryBlock.getOutputExpression(), queryBlock.getOrderByAnalytic()),
-                        queryBlock.getOutputExprInOrderByScope(),
-                        queryBlock.getColumnOutputNames(),
-                        builder.getFieldMappings(),
-                        queryBlock.getOrderScope(), false);
-            } else {
-                //requires output fields, groups and translated aggregations to be visible for queries with aggregation
+            if (!queryBlock.getGroupBy().isEmpty() || !queryBlock.getAggregate().isEmpty()) {
```

Comment:
sr can work well without sql_mode `MODE_ONLY_FULL_GROUP_BY`, will effect this case?

Reply:
no effect. If `MODE_ONLY_FULL_GROUP_BY` works, it means there exisit group by clause. This pr works as the old in this scene.